wormhole-connect/src/utils/vaa.ts

```diff
@@ -37,6 +39,140 @@ export type MessageIdentifier = {
+      '0xf93124b7c738843cbb89e864c862c38cddcccf95',
+      '0xd2cc37a4dc036a8d232b48f62cdd4731412f4890',
+      '0xda798f6896a3331f64b48c12d1d57fd9cbe70811',
+      '0x71aa1be1d36cafe3867910f99c09e347899c19c3',
+      '0x8192b6e7387ccd768277c17dab1b7a5027c0b3cf',
+      '0x178e21ad2e77ae06711549cfbb1f9c7a9d8096e8',
+      '0x5e1487f35515d02a92753504a8d75471b9f49edb',
+      '0x6fbebc898f403e4773e95feb15e80c9a99c8348d',
+    ],
+    expiry: 0,
+  },
+  testnet: {
+    index: 0,
+    keys: ['0x13947Bd48b18E53fdAeEe77F3473391aC727C638'],
+    expiry: 0,
+  },
+};
+
+type HomogeneousFunction<T, Args extends any[]> = (
+  ...args: Args
+) => T | undefined;
+
+// Create an executor that uses homogeneous function types and handles arbitrary arguments
+const createCustomExecutor =
+  <T, Args extends any[]>(
+    ...funcs: Array<HomogeneousFunction<T, Args>>
+  ): ((...args: Args) => Promise<T | undefined>) =>
+  async (...args: Args): Promise<T | undefined> => {
+    for (let i = 0; i < funcs.length; i++) {
+      const result = await funcs[i](...args);
+      if (result !== undefined) {
+        return result;
+      }
+    }
+  };
+
+const fetchSignedVaaFromGuardian =
+  (host: string) =>
+  (emitterChain: ChainId, emitterAddress: string, sequence: string) =>
+    getSignedVAA(host, emitterChain, emitterAddress, sequence);
+
+const guardianVaaFetchers = config.wormholeRpcHosts.map((host) =>
+  fetchSignedVaaFromGuardian(host),
+);
+
+async function fetchVaaBytesFromGuardian(
+  txData: ParsedMessage | ParsedRelayerMessage,
+): Promise<Uint8Array | undefined> {
+  try {
+    const messageId = getEmitterAndSequence(txData);
+    const { emitterChain, emitterAddress, sequence } = messageId;
+    // round-robin through the RPC hosts
+    const fetchVaa = createCustomExecutor(...guardianVaaFetchers);
+    const result = await fetchVaa(emitterChain, emitterAddress, sequence);
+    if (result && result.vaaBytes) {
+      return result.vaaBytes;
+    }
+  } catch (e: any) {
+    //on error, log it for debugging
+    console.error('Failed to fetch VAA from Guardians', e);
+  }
+}
+
+async function fetchVaaBytesFromWormholescan(
+  txData: ParsedMessage | ParsedRelayerMessage,
+): Promise<Uint8Array | undefined> {
+  try {
+    const messageId = getEmitterAndSequence(txData);
+    const { emitterChain, emitterAddress, sequence } = messageId;
+    const url = `${config.wormholeApi}api/v1/vaas/${emitterChain}/${emitterAddress}/${sequence}`;
+    const response = await axios.get(url);
+    if (response.data.data) {
+      const data = response.data.data;
+      return utils.base64.decode(data.vaa);
+    }
+  } catch (e: any) {
+    //on error, log it for debugging
+    console.error('Failed to fetch VAA from Wormholescan', e);
+  }
+}
+
+type VaaParseInput = {
+  vaa: Uint8Array;
+  txData: ParsedMessage | ParsedRelayerMessage;
+};
+
+function parseVaa({ vaa, txData }: VaaParseInput) {
+  const parsed = parseTokenTransferVaa(vaa);
```

Comment:
This might not be a token transfer vaa

Reply:
I didn't this logic, I've just moved, it was strage to me as well